### ISSUE TITLE
master: jquery-ui must be fully synchronous

### ISSUE THREAD

**Issue by mrclay** [Elgg/Elgg#8515]
The solution in #8484 does not cut it (try the forms theme_sandbox page). We need UI's factory function to run fully synchronously. This seems to work:

```php
<?php
$js = file_get_contents(elgg_get_root_path() . "/vendor/bower-asset/jquery-ui/jquery-ui.min.js");

// We must hide define() from jQuery UI in order to reliably run its factory
// function synchronously. If we let it define(), require() does not block and
// allows other scripts to run code before $.ui is available.
?>
//<script>
!function(define) {
	<?= $js ?>
}();
define('jquery-ui', null);
```

**Comment by mrclay** [comment]
I think `js/elgg` and the languages file needs to be sync as well. This would get rid of these intermittent problems where RequireJS tries to load elgg.js as a module.

**Comment by mrclay** [comment]
This fix is in #8513.
